shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz_hints/domain/model/HintState.kt

```diff
@@ -1,10 +1,25 @@
 package org.hyperskill.app.step_quiz_hints.domain.model
 
 import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.jsonPrimitive
 import org.hyperskill.app.user_storage.domain.model.UserStorageValue
 
 enum class HintState(val userStorageValue: UserStorageValue) {
     SEEN(JsonPrimitive("seen")),
     HELPFUL(JsonPrimitive("helpful")),
-    UNHELPFUL(JsonPrimitive("unhelpful")),
+    UNHELPFUL(JsonPrimitive("unhelpful"));
+
+    companion object {
+        private val VALUES: Array<HintState> = values()
+
+        fun getByStringValue(value: String): HintState? =
+            VALUES.firstOrNull { it.userStorageValue.jsonPrimitive.content == value }
+    }
```

Comment:
enums can be automatically serialized using `kotlinx.serialization`. You don't need to use `JsonProomotive`. 
In this case you just need to mark each invariant with `@SerialName("originName")` annotation.
See documentation https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/builtin-classes.md#enum-classes.

Reply:
Also, if it doesn't matter which name is used for `enum` entry, you can use built-in name. 
See documentation https://kotlinlang.org/docs/enum-classes.html#working-with-enum-constants

Reply:
The main usage of the `HintState` is in `StepQuizHintsActionDispatcher` when updating user storage via `UserStorageInteractor`. The `UserStorageInteractor` API works with the JSON types directly (see `/user_storage/domain/model/UserStorage.kt`). So it's currently impossible to implement the proposed changes and it's out of the scope of this PR.